dexlib2/src/main/java/org/jf/dexlib2/util/Preconditions.java

```diff
@@ -188,6 +195,48 @@ public static int checkVerificationError(int verificationError) {
         return verificationError;
     }
 
+    public static ImmutableList<? extends ImmutableSwitchElement> checkSequentialKeys(ImmutableList<? extends ImmutableSwitchElement> elements) {
```

Comment:
It would be good if this was a template method with a parameter that extends something like List<? extends SwitchElement> for the argument and the return type, so it's not restricted to just the Immutable* implementation.

Reply:
I've tried to maintain your style of all check methods beeing identity methods.

We could try to generify the method
```java
public static <E extends SwitchElement, C extends Collection<E>> C checkSequentialKeys(C elements)   { ... }
```

Or we just don't return the parameter and call the method normally instead of nesting it:
```java
public static void checkSequentialKeys(List<? extends SwitchElement> elements) { ... }
```
```java
public ImmutablePackedSwitchPayload(
        @Nullable ImmutableList<? extends ImmutableSwitchElement> switchElements) {
    super(OPCODE);
    this.switchElements = ImmutableUtils.nullToEmptyList(switchElements);
    Preconditions.checkSequentialKeys(this.switchElements);
}
```

Reply:
Yeah, your first example is what I had in mind

```java
public static <E extends SwitchElement, C extends Collection<E>> C checkSequentialKeys(C elements)   { ... }
```